### PR TITLE
Hotfix/sudoku input

### PR DESCRIPTION
**Pull request by patrikglanemann**
fixes #15 

What I did:
- Add uuid
- Save initial Sudoku in local storage 
    - Use it as a mask to determine if a cell is editable or not
    - Add logic to disable button (cell) if cell is not editable
- Add if statement in handleNumberInputClick function to check if a cell got selected

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,39 +2,58 @@ import "./App.css";
...
       .then((data) => {
         setNumbers(data.board);
+        localStorage.setItem("currentSudoku", JSON.stringify(data.board));
+        setIsLoading(false);
       });
   }, []);
 
   function renderCellRow(rowNumber) {
-    const numbArray = numbers[rowNumber].map((number, index) => {
+    if (isLoading || numbers === null || numbers === undefined) {
```
Comment (johnxgaehner) [inline src/App.js:25]: you could shorten this a bit and check for `(isLoading || !numbers)`, right?

File: src/App.js [hunk 2]
```diff
@@ -2,39 +2,58 @@ import "./App.css";
...
       return (
-        <Cell
-          key={index}
-          value={number}
-          id={[rowNumber, index]}
-          onClick={handleCellClick}
-        />
+        <>
+          <p>Loading...</p>
+        </>
```
Comment (johnxgaehner) [inline src/App.js:29]: are the fragments necessary here?

**Review by johnxgaehner — COMMENTED** [review]
great work! bugs are fixed on my side.
also a very clean written code, nice to read. :)

File: src/App.js [hunk 3]
```diff
@@ -2,39 +2,58 @@ import "./App.css";
...
       );
-    });
-    return numbArray;
+    } else {
+      let mask = JSON.parse(localStorage.getItem("currentSudoku"));
+      console.log("mask: " + mask);
```
Comment (johnxgaehner) [inline src/App.js:33]: I was told that we should remove console logs in the final code. ;-)))

File: src/App.js (hunk 1 above)
Comment (patrikglanemann) [reply src/App.js:25]: True! Thank for pointing out.

File: src/App.js (hunk 2 above)
Comment (patrikglanemann) [reply src/App.js:29]: I tried without them and nope they are not needed:D

File: src/App.js (hunk 3 above)
Comment (patrikglanemann) [reply src/App.js:33]: Ah shoot, I copy pasted it back in XD

**Pull request merged**